ci/docker_cache.py

```diff
@@ -28,237 +28,164 @@
+    _login_dockerhub()
+    # We don't have to retag the image since it is already in the right format
+    logging.info('Uploading %s (%s) to %s', docker_tag, image_id, registry)
+    push_cmd = ['docker', 'push', docker_tag]
+    subprocess.check_call(push_cmd)
 
-    # Compile layers into tarfile
-    with tempfile.TemporaryDirectory() as temp_dir:
-        tar_file_path = _format_docker_cache_filepath(output_dir=temp_dir, docker_tag=docker_tag)
-        logging.debug('Writing layers of {} to {}'.format(docker_tag, tar_file_path))
-        history_cmd = ['docker', 'history', '-q', docker_tag]
 
-        image_ids_b = subprocess.check_output(history_cmd)
-        image_ids_str = image_ids_b.decode('utf-8').strip()
-        layer_ids = [id.strip() for id in image_ids_str.split('\n') if id != '<missing>']
-
-        # docker_tag is important to preserve the image name. Otherwise, the --cache-from feature will not be able to
-        # reference the loaded cache later on. The other layer ids are added to ensure all intermediary layers
-        # are preserved to allow resuming the cache at any point
-        cmd = ['docker', 'save', '-o', tar_file_path, docker_tag]
-        cmd.extend(layer_ids)
-        try:
-            check_call(cmd)
-        except CalledProcessError as e:
-            logging.error('Error during save of {} at {}. Command: {}'.
-                          format(docker_tag, tar_file_path, pprint.pprint(cmd)))
-            return
-
-        # Upload file
-        logging.info('Uploading {} to S3'.format(docker_tag))
-        with open(tar_file_path, 'rb') as data:
-            s3_object.upload_fileobj(
-                Fileobj=data,
-                Callback=ProgressPercentage(object_name=docker_tag, size=os.path.getsize(tar_file_path)),
-                ExtraArgs={"Metadata": {S3_METADATA_IMAGE_ID_KEY: image_id}})
-            logging.info('Uploaded {} to S3'.format(docker_tag))
-
-
-def _get_remote_image_id(s3_object) -> str:
+def _login_dockerhub():
     """
-    Get the image id of the docker cache which is represented by the S3 object
-    :param s3_object: S3 object
-    :return: Image id as string or None if object does not exist
+    Login to the Docker Hub account
+    :return: None
     """
-    try:
-        if S3_METADATA_IMAGE_ID_KEY in s3_object.metadata:
-            cached_image_id = s3_object.metadata[S3_METADATA_IMAGE_ID_KEY]
-            return cached_image_id
-        else:
-            logging.debug('No cached image available for {}'.format(s3_object.key))
-    except botocore.exceptions.ClientError as e:
-        if e.response['Error']['Code'] == "404":
-            logging.debug('{} does not exist in S3 yet'.format(s3_object.key))
-        else:
-            raise
+    dockerhub_credentials = _get_dockerhub_credentials()
+    login_cmd = ['docker', 'login', '--username', dockerhub_credentials['username'], '--password',
+                 dockerhub_credentials['password']]
```

Comment:
would anyone be able to update this file and print out the password in the log?

Reply:
no, this only runs on restricted slaves which are not executing code from PRs. On regular slaves, this password is not accessible